src/components/QueryEditor.tsx

```diff
@@ -0,0 +1,204 @@
+  }))
+  .sortBy('label')
+  .value();
+
+export function QueryEditor({ query, onChange, onRunQuery }: Props) {
+  const [tagName, setTagName] = useState<string | null>(null);
+  const [tagValue, setTagValue] = useState('');
+  const theme = useTheme2();
+
+  const debouncedOnRunQuery = useMemo(() => _.debounce(onRunQuery, 500), [onRunQuery]);
+
+  const onQueryTextChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange({ ...query, queryText: event.target.value });
+    debouncedOnRunQuery();
+  };
+
+  const onRealmChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange({ ...query, realm: event.target.value });
+    debouncedOnRunQuery();
+  };
+
+  const onMetricChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange({ ...query, metric: event.target.value });
+    debouncedOnRunQuery();
+  };
+
+  const onAliasChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange({ ...query, alias: event.target.value });
+    debouncedOnRunQuery();
+  };
+
+  const onTagNameChange = (option: SelectableValue<string>) => {
+    if (option.value) {
+      setTagName(option.value);
+    }
+  };
+
+  const onTagValueChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTagValue(event.target.value);
+  };
+
+  const onAddTag = () => {
+    if (tagName) {
+      onChange({ ...query, tags: { ...query.tags, [tagKeyLookup[tagName] ?? tagName]: tagValue } });
+      setTagName(null);
+      setTagValue('');
+      debouncedOnRunQuery();
+    }
+  };
+
+  const onRemoveTag = (name: string) => {
+    onChange({ ...query, tags: _.omit(query.tags, name) });
+    debouncedOnRunQuery();
+  };
+
+  const onDownsamplerChange = (option: SelectableValue<string>) => {
+    onChange({ ...query, ...(option.value !== DEFAULT_DOWNSAMPLER && { downsampler: option.value }) });
+    debouncedOnRunQuery();
+  };
+
+  const { queryText, realm, metric, tags, downsampler, alias } = query;
+  const areControlsDisabled = Boolean(queryText);
+
+  return (
+    <div>
+      <InlineField label="Realm" labelWidth={16} required>
+        <Input onChange={onRealmChange} value={realm} width={30} />
+      </InlineField>
+      <InlineField label="Alias" labelWidth={16}>
+        <Input onChange={onAliasChange} value={alias} width={30} />
+      </InlineField>
+      <InlineField
+        label="Query Text"
+        labelWidth={16}
+        tooltip="If you know the metrics API query
+        string you can paste it here, e.g. m=tasks&u=bob&downsampler=maximum. Otherwise, use the fields
+        below to build your query."
+      >
+        <Input onChange={onQueryTextChange} value={queryText || ''} width={60} />
+      </InlineField>
+      <InlineField
+        label="Metric name"
+        labelWidth={16}
+        disabled={areControlsDisabled}
+        tooltip="Metric urlname. Example: tasks, rssram. Simple operations can be performed on metrics using
+            RPN notation. For example if you wanted to divide the ram per task you would do:
```

Comment:
maybe link to https://dashboard.pepperdata.com/{realm}/metricdefinitions to get the metrics?